testsuite/integration/elytron-oidc-client/src/test/java/org/wildfly/test/integration/elytron/oidc/client/OidcBaseTest.java

```diff
@@ -0,0 +1,300 @@
+import org.apache.http.protocol.BasicHttpContext;
+import org.apache.http.protocol.HttpContext;
+import org.jboss.arquillian.container.test.api.OperateOnDeployment;
+import org.jboss.as.arquillian.api.ServerSetupTask;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.test.http.util.TestHttpClientUtils;
+import org.jboss.as.test.integration.security.common.servlets.SimpleSecuredServlet;
+import org.jboss.as.test.integration.security.common.servlets.SimpleServlet;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.testcontainers.DockerClientFactory;
+import org.wildfly.security.jose.util.JsonSerialization;
+
+import io.restassured.RestAssured;
+
+/**
+ * Tests for the OpenID Connect authentication mechanism.
+ *
+ * @author <a href="mailto:fjuma@redhat.com">Farah Juma</a>
+ */
+public abstract class OidcBaseTest {
+
+    public static final String CLIENT_SECRET = "secret";
+    public static final String OIDC_WITHOUT_SUBSYSTEM_CONFIG_WEB_XML = "web.xml";
+    public static KeycloakContainer KEYCLOAK_CONTAINER;
+    public static final String TEST_REALM = "WildFly";
+    private static final String KEYCLOAK_USERNAME = "username";
+    private static final String KEYCLOAK_PASSWORD = "password";
+    public static final int CLIENT_PORT = TestSuiteEnvironment.getHttpPort();
+    public static final String CLIENT_HOST_NAME = TestSuiteEnvironment.getHttpAddress();
+    public static final String PROVIDER_URL_APP = "ProviderUrlOidcApp";
+    public static final String AUTH_SERVER_URL_APP = "AuthServerUrlOidcApp";
+    public static final String WRONG_PROVIDER_URL_APP = "WrongProviderUrlOidcApp";
+    public static final String WRONG_SECRET_APP = "WrongSecretOidcApp";
+
+    private static boolean isDockerAvailable() {
+        try {
+            DockerClientFactory.instance().client();
+            return true;
+        } catch (Throwable ex) {
+            return false;
+        }
+    }
+
+    public static void sendRealmCreationRequest(RealmRepresentation realm) {
+        try {
+            RestAssured
+                    .given()
+                    .auth().oauth2(KeycloakConfiguration.getAdminAccessToken(KEYCLOAK_CONTAINER.getAuthServerUrl()))
+                    .contentType("application/json")
+                    .body(JsonSerialization.writeValueAsBytes(realm))
+                    .when()
+                    .post(KEYCLOAK_CONTAINER.getAuthServerUrl() + "/admin/realms").then()
+                    .statusCode(201);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @BeforeClass
+    public static void checkDockerAvailability() {
+        assumeTrue("Docker isn't available, OIDC tests will be skipped", isDockerAvailable());
```

Comment:
This doesn't work for OidcWithDeploymentConfigTest 

https://ci.wildfly.org/viewLog.html?buildId=272892&tab=buildResultsDiv&buildTypeId=WF_PullRequest_WindowsJdk11#testNameId2612006752933148236

I think it is somehow because Arq deploys archives produced by Deployment methods before executing @BeforeClass methods. That said though I do not understand why KeycloakAndSystemPropertySetup wouldn't be executed before the deployment. It would be surprising if a ServerSetup wasn't executed before deployment, as we surely have many deployments in our TS that won't work if the ServerSetup hasn't been run.

Reply:
@Skyllarr Your comment above to Darran re https://github.com/wildfly/wildfly-arquillian/blob/main/common/src/main/java/org/jboss/as/arquillian/container/ServerSetupObserver.java#L187 explains it. The ServerSetup fails and the failure is swallowed. Then the deployment fails.

Reply:
I'm going to merge this and will then either send up a hack to try and fix this or will @Ignore the test.

Reply:
https://issues.redhat.com/browse/WFLY-15297 is to track fixing this. I'll @Ignore the test now so failures don't disrupt others' work.

Reply:
@bstansberry Thank you